### ISSUE TITLE
Make `BitWidth` use `NonZeroUsize`

### ISSUE THREAD

**Issue by AaronKutch** [Robbepop/apint#39]
The compiler has special optimizations for this. In fact, I might just replace `BitWidth` altogether, but I will have to try that in an experimental branch in the future.

**Comment by Robbepop** [comment]
I was also already thinking about this and also to use other things such as `MaybeUninit`, `ManuallyDrop`. I think `apint` could make use a lot more of those new abstractions.

**Comment by AaronKutch** [comment]
Yeah. I learned a bunch about manipulating pointers and using `MaybeUninit` when working on my `ptr_rotate` PR, so I will do this.

**Comment by AaronKutch** [comment]
I was considering replacing `BitWidth` altogether with `NonZeroUsize`, but I have decided to keep it as a wrapper type around a `NonZeroUsize`. I found it unlikely that we would use unused bits in the `len`, but someone might want to clone the library and use `NonZeroU32` or something else special.
